### PR TITLE
Add recipe for MELODIES MONET

### PR DESCRIPTION
**Pull request by zmoon**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

cc: @rschwant

### CODE REVIEW

**Review by rschwant — APPROVED** [review]
This looks great to me! And I like the idea of leaving wrf-python out by default and users can add it in if they want to run the wrf-chem scripts and they are running on an applicable machine.

**Comment by zmoon** [comment]
@conda-forge/help-python

File: recipes/melodies-monet/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/NOAA-CSL/MELODIES-MONET/archive/v{{ version }}.tar.gz
+  sha256: 11348c50e447bb54e6a9f8c8ead71c6ee748d97f58b532efbb0d098f9b882ecd
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```
Comment (ocefpaf) [inline recipes/melodies-monet/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
```

File: recipes/melodies-monet/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{% set name = "melodies-monet" %}
+{% set version = "0.1.dev1" %}
```
Comment (ocefpaf) [inline recipes/melodies-monet/meta.yaml:2]: This one falls under https://github.com/conda-forge/cfep/blob/main/cfep-05.md

File: recipes/melodies-monet/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+source:
+  url: https://github.com/NOAA-CSL/MELODIES-MONET/archive/v{{ version }}.tar.gz
+  sha256: 11348c50e447bb54e6a9f8c8ead71c6ee748d97f58b532efbb0d098f9b882ecd
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  entry_points:
+    - "melodies-monet = melodies_monet._cli:cli"
```
Comment (ocefpaf) [inline recipes/melodies-monet/meta.yaml:17]:
```suggestion
    - melodies-monet = melodies_monet._cli:cli
```

File: recipes/melodies-monet/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+requirements:
+  host:
+    - python >=3.7
+    - pip
+    - setuptools
+  run:
+    - python >=3.7
+    #
```
Comment (ocefpaf) [inline recipes/melodies-monet/meta.yaml:26]: Let's save a few lines. The comment below is good enough to separate the sections.
```suggestion
```

File: recipes/melodies-monet/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    - cartopy
+    - matplotlib-base <3.9  # pandas v1 compat
+    - monet
+    - monetio
+    - numpy
+    - pandas <2  # some monetio obs readers fail with pandas v2
+    - pyyaml
+    - seaborn
+    - xarray
+    #
```
Comment (ocefpaf) [inline recipes/melodies-monet/meta.yaml:37]:
```suggestion
```

File: recipes/melodies-monet/meta.yaml (hunk 2 above)
Comment (zmoon) [reply recipes/melodies-monet/meta.yaml:2]: If I'm reading that correctly, it just means that the following needs to be used in order for the install to work (until there's a stable release)?
```
conda install -c conda-forge/label/melodies-monet_dev -c conda-forge melodies-monet
```

Comment (ocefpaf) [reply recipes/melodies-monet/meta.yaml:2]: That is for the user. You would need to add a `conda_build_config.yaml` here according to https://github.com/conda-forge/cfep/blob/main/cfep-05.md#specification.

If you can request a stable release upstream, you don't need that step.

**Comment by zmoon** [comment]
@conda-forge/help-python please re-review

File: recipes/melodies-monet/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    # Extras
+    - dask
+    - joblib
+    - netcdf4
+    - pooch
+    - pytz
+    - rich
+    - timezonefinder
+    - typer
+    # - wrf-python  # for WRF-Chem reader
```
[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/melodies-monet/meta.yaml:45]: Is `wrf-python` an optional dependency? If it is a mandatory dependency that is not advisable. If it is optional and you want to include in selected archs, see https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-packages-with-os-specific-dependencies.

[the remark this answers is not in the document]
Comment (zmoon) [reply recipes/melodies-monet/meta.yaml:45]: Yeah it's just optional.

[the remark this answers is not in the document]
Comment (zmoon) [reply recipes/melodies-monet/meta.yaml:45]: Seems like it'll be easier just to wait until wrf-python feedstock builds for Apple Silicon and Python 3.12.

**Comment by zmoon** [comment]
Thanks @ocefpaf for your help!

**Pull request merged**